packages/cli/fragments/fragment-vanilla-ts/src/main.ts

```diff
@@ -0,0 +1,25 @@
+import { invoke } from "@tauri-apps/api";
+
+declare global {
+  interface Window {
+    greet: () => void;
```

Comment:
add type definitions for `window.__TAURI__`

Reply:
You want me to redeclare all functions types inside window.__TAURI__ ?

Reply:
no, just nivoke

Reply:
What about using `@tauri-apps/api` as `dev` dependency and do something like this:
```ts
import type * as TauriApiTypes from "@tauri-apps/api";

declare global {
  interface Window {
    greet: () => void;
    __TAURI__: typeof TauriApiTypes
  }
}

const { invoke } = window.__TAURI__;

let greetInputEl: HTMLInputElement | null;
let greetMsgEl: HTMLElement | null;

window.addEventListener("DOMContentLoaded", () => {
  greetInputEl = document.querySelector("#greet-input");
  greetMsgEl = document.querySelector("#greet-msg");
});

async function greet() {
  if (greetMsgEl && greetInputEl) {
    greetMsgEl.textContent = await invoke("greet", {
      name: greetInputEl.value,
    });
  }
}

window.greet = greet;
```

Reply:
I like that, but could you extract that in a shim file outside `main.ts` ?

Reply:
something like this https://github.com/vitejs/vite/blob/main/packages/create-vite/template-vue-ts/src/vite-env.d.ts

Reply:
Yeah sure sounds good to me i have updated.